pkg/apis/ark/v1/backup_storage_location.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupStorageLocation is a location where Ark stores backup objects.
+type BackupStorageLocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   BackupStorageLocationSpec   `json:"spec"`
+	Status BackupStorageLocationStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupStorageLocationList is a list of BackupStorageLocations.
+type BackupStorageLocationList struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Items             []BackupStorageLocation `json:"items"`
+}
+
+// StorageType represents the type of storage that a backup location uses.
+// Exactly one must be set.
```

Comment:
Making this comment with trepidation due to low confidence I'm getting this right: is it possible to set more than one `StorageType`? If not, this comment would be confusing.

Reply:
currently, yes. it's set up to be extensible over time for other types of backends besides object storage to be added. fine to make the comment explicit about that, i.e. "ObjectStorage must be non-nil as it's currently the only supported StorageType"

Reply:
I like that ⬆️ and modifying it to "Exactly one of its members must be set." when another member is added.